regression/test.pl

```diff
@@ -115,12 +112,44 @@ ($$$$$)
         last if($included == -1);
         if($result eq "--") {
           $included--;
-        } else {
+        } elsif($result !~ /\s*/) {
           my $r;
-          $result =~ s/\\/\\\\/g;
-          $result =~ s/([^\\])\$/$1\\r\\\\?\$/;
-          system("bash", "-c", "grep $grep_options \$'$result' '$name/$output' >/dev/null");
-          $r = ($included ? $? != 0 : $? == 0);
+
+          my $dir = getcwd();
+          my $abs_path = "$dir/$name/$output";
+          open( my $fh => $abs_path) || die "Cannot open '$name/$output': $!";
+
+          # Multi line therefore we run each check across the whole output
+          if($grep_options eq "activate-multi-line-match") {
+            local $/ = undef;
+            binmode $fh;
+            my $whole_file = <$fh>;
+            my $is_match = $whole_file =~ /$result/;
+            $r = ($included ? !$is_match : $is_match);
+          }
+          else
+          {
+            my $found_line = 0;
+            while(my $line = <$fh>) {
+              if($line =~ /$result/) {
+                # We've found the line, therefore if it is included we set the result to 0 (OK)
+                # If it is excluded, we set the result to 1 (FAILED)
+                $r = !$included;
+                $found_line = 1;
+                last;
+              }
+            }
+
+            if($found_line == 0) {
+              # None of the lines matched, therefore the result is set to 0 (OK) iff the line
```

Comment:
No, iff is short hand for "if and only" if but that's silly since not widely known, I will correct. In fact perhaps the only if condition doesn't really add much.